platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpClientBuilder.java

```diff
@@ -0,0 +1,200 @@
+
+      String[] defaultCipherSuites =
+          AccessController.doPrivileged(
+              (PrivilegedAction<String[]>)
+                  () -> commaSeparatedToArray(System.getProperty(HTTPS_CIPHER_SUITES)));
+
+      httpClientBuilder.setSSLSocketFactory(
+          new SSLConnectionSocketFactory(
+              getSslContext(),
+              defaultProtocols,
+              defaultCipherSuites,
+              SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER));
+    }
+    if (isConfiguredForBasicAuth()) {
+      httpClientBuilder.setDefaultCredentialsProvider(getCredentialsProvider());
+      httpClientBuilder.addInterceptorFirst(new PreemptiveAuth(new BasicScheme()));
+    }
+    return httpClientBuilder;
+  }
+
+  private Boolean isConfiguredForBasicAuth() {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<Boolean>) () -> Boolean.valueOf(System.getProperty("solr.useBasicAuth")));
+  }
+
+  public boolean useTls() {
+    return StringUtils.startsWithIgnoreCase(
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
+        "https");
+  }
+
+  private CredentialsProvider getCredentialsProvider() {
+    CredentialsProvider provider = new BasicCredentialsProvider();
+
+    org.apache.http.auth.UsernamePasswordCredentials credentials =
+        new org.apache.http.auth.UsernamePasswordCredentials(
+            AccessController.doPrivileged(
+                (PrivilegedAction<String>) () -> System.getProperty("solr.username")),
+            getPlainTextSolrPassword());
+    provider.setCredentials(AuthScope.ANY, credentials);
+    return provider;
+  }
+
+  private static SSLContext getSslContext() {
+    final Boolean check =
+        AccessController.doPrivileged(
+            (PrivilegedAction<Boolean>)
+                () ->
+                    (System.getProperty(KEY_STORE) == null
+                        || System.getProperty(KEY_STORE_PASS) == null
+                        || System.getProperty(TRUST_STORE) == null
+                        || System.getProperty(TRUST_STORE_PASS) == null));
+
+    if (check) {
+      throw new IllegalArgumentException("KeyStore and TrustStore system properties must be set.");
+    }
+
+    final KeyStore[] trustStore = new KeyStore[1];
+    final KeyStore[] keyStore = new KeyStore[1];
+
+    AccessController.doPrivileged(
+        (PrivilegedAction<Object>)
+            () -> {
+              trustStore[0] =
+                  getKeyStore(
+                      System.getProperty(TRUST_STORE), System.getProperty(TRUST_STORE_PASS));
+              keyStore[0] =
+                  getKeyStore(System.getProperty(KEY_STORE), System.getProperty(KEY_STORE_PASS));
+              return null;
+            });
+
+    // IDE thinks this variable does not need to be initialized. However, because of the try/catch
+    // block, there is a situation where it could end up uninitialized.
```

Comment:
I'd probably remove this comment

Reply:
That's fine. I added it because I removed the null initialized on a previous PR and my PR was X'ed by a reviewer.